src/main/java/com/ericsson/ei/handlers/MongoDBHandler.java

```diff
@@ -266,8 +268,46 @@ public void createTTLIndex(String dataBaseName, String collectionName, String fi
     private MongoCollection<Document> getMongoCollection(String dataBaseName, String collectionName) {
         if (mongoClient == null)
             return null;
-        MongoDatabase db = mongoClient.getDatabase(dataBaseName);
-        List<String> collectionList = db.listCollectionNames().into(new ArrayList<String>());
+        MongoDatabase db;
+        List<String> collectionList;
+        try {
+            db = mongoClient.getDatabase(dataBaseName);
+            collectionList = db.listCollectionNames().into(new ArrayList<String>());
+        }
+        catch (MongoCommandException e) {
+                LOGGER.error("MongoCommandException, Something went wrong with MongoDb connection. Error: " + e.getErrorMessage() + "\nStacktrace\n" + e);
+                if (mongoClient != null) {
```

Comment:
closing mongoClient and setting it to null should be done in a separate method to avoid all the copied code

Reply:
fixed.